ugvc/cnv/convert_cnv_results_to_vcf.py

```diff
@@ -0,0 +1,156 @@
+    header.add_line('##INFO=<ID=CopyNumber,Number=1,Type=Float,Description="copy number of CNV call">')
+    header.add_line('##INFO=<ID=RoundedCopyNumber,Number=1,Type=Integer,Description="rounded copy number of CNV call">')
+    #header.add_line('##INFO=<ID=END_POS,Description="end position of the CNV">')
+    header.add_line('##INFO=<ID=SVLEN,Number=.,Type=Integer,Description="CNV length">')
+    header.add_line('##INFO=<ID=SVTYPE,Number=1,Type=String,Description="CNV type. can be DUP or DEL">')
+    
+    # Add FORMAT
+    header.add_line('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">')
+
+    # Open a VCF file for writing
+    if args.out_directory: 
+        out_directory=args.out_directory
+    else:
+        out_directory=""
+    outfile=pjoin(out_directory,sample_name+'.cnv.vcf')
+    
+    with pysam.VariantFile(outfile, mode='w', header=header) as vcf_out:
+        df_cnvs = pd.read_csv(args.cnv_annotated_bed_file,sep="\t",header=None)
+        df_cnvs.columns=['chr','start','end','info']
+        for index, row in df_cnvs.iterrows():
+            # Create a new VCF record
+            chrID = row['chr']
+            start = row['start']
+            end = row['end']
+            info = row['info']
+
+            CN = int(info.split("|")[0].replace("CN",""))
+            cnv_type = '<DUP>'
+            if CN<2:
+                cnv_type='<DEL>'
+
+            filters = []
+            for item in info.split(';'):
+                arr = item.split('|')
+                if len(arr)>1:
+                    filters.append(arr[1])
+            
+            record = vcf_out.new_record()
+            record.contig =  chrID
+            record.start = start
+            record.stop = end
+            record.ref = "N"
+            record.alts = (cnv_type,)
+
+            CONFIDENCE = 'HIGH'
+            if len(filters)>0 :
+                for f in filters:
+                    record.filter.add(f)
+                CONFIDENCE='MEDIUM'
+                if len(filters)>1 :
+                    CONFIDENCE='LOW'
+            else:
+                record.filter.add('PASS')
+          
+            record.info['CONFIDENCE'] = CONFIDENCE
+            record.info['CopyNumber'] = CN
+            record.info['RoundedCopyNumber'] = int(CN)
+            record.info['SVLEN'] = int(end)-int(start)
+            record.info['SVTYPE'] = cnv_type.replace("<","").replace(">","")
+            #record.info['END_POS'] = str(end)
+
+
+            # Set genotype information for each sample
+            GT=[None,1]
+            if CN==1:
+                GT=[0,1]
+            elif CN==0:
+                GT=[1,1]
+            record.samples[sample_name]['GT'] = (GT[0],GT[1]) 
+
+            # Write the record to the VCF file
+            vcf_out.write(record)
+        
+        vcf_out.close()
+        cmd = f"bgzip -c {outfile} > {outfile}.gz" 
```

Comment:
I think that if the file name that you give pysam is vcf.gz, the pysam writes a gzipped file.